AppController/djinn.rb

```diff
@@ -754,12 +754,13 @@ def check_layout(layout, keyname)
         raise AppScaleException.new(msg)
       end
       if !node['public_ip'] || !node['private_ip'] || !node['jobs'] ||
-        !node['instance_id']
+        !node['instance_id'] || !node['instance_type']
         msg = "Error: node layout is missing information #{node}."
         Djinn.log_error(msg)
         raise AppScaleException.new(msg)
       elsif node['public_ip'].empty? || node['private_ip'].empty? ||
-         node['jobs'].empty? || node['instance_id'].empty?
+         node['jobs'].empty? || node['instance_id'].empty? ||
+         node['instance_type'].empty?
```

Comment:
Is there any change in the AppScalefile for the new type?

Reply:
Yes, we would now specify the different instance types within the roles section of the ips layout.